modules/core/src/channels.cpp

```diff
@@ -53,25 +53,25 @@ static void mixChannels8u( const uchar** src, const int* sdelta,
     mixChannels_(src, sdelta, dst, ddelta, len, npairs);
 }
 
-static void mixChannels16u( const ushort** src, const int* sdelta,
-                            ushort** dst, const int* ddelta,
+static void mixChannels16u( const uchar** src, const int* sdelta,
+                            uchar** dst, const int* ddelta,
                             int len, int npairs )
 {
-    mixChannels_(src, sdelta, dst, ddelta, len, npairs);
+    mixChannels_((const ushort**)src, sdelta, (ushort**)dst, ddelta, len, npairs);
 }
 
-static void mixChannels32s( const int** src, const int* sdelta,
-                            int** dst, const int* ddelta,
+static void mixChannels32s( const uchar** src, const int* sdelta,
+                            uchar** dst, const int* ddelta,
```

Comment:
It is better to use `void**` as buffer type is not specified in the interface (implementation-defined).